autolev: removed numpy and math imports from parser outputs

sudz123 opened the pull request:
Fixes #15186 

#### Release Notes
<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

sudz123 commented:
Oh right, sorry I just realised that. I'll remove this PR and put up a new one

NikhilPappu commented:
@sudz123 Please don't create duplicate PRs! Make the changes here.

sudz123 commented:
@NikhilPappu so I should remove this line: `self.output_code.append("import math as m\n")` and for the numpy import, where and how should the flag be declared? I am new to this and still trying to get familiar with the code, so thanks for helping out.

sudz123 commented:
I already closed this one, I will be careful in the future, is there no way to re-open this?

NikhilPappu commented:
@sudz123 You should modify the import lines like you mentioned and also all the places where it is referenced. Just make a boolean value in the Listener class and set it to False by default. Pass this as a parameter to the `parse_autolev` function in `_parse_autolev_antlr.py` and `__init__.py` so users can set it.

sudz123 committed:
autolev: added flag numpy_import in class
MyListener

Default is set to False. Can be set to
True in case numpy import is necessary.

sudz123 committed:
Removed wrapper_module_8.pyx

NikhilPappu commented:
@sudz123 Its not quite there. The flag isn't doing anything right now. You need to look at all the strings which have `np.` and `m.`. The parser basically writes these outputs. You should remove anything that uses `.m` as this import is removed. For `np.` you need to use the flag to include it.

NikhilPappu commented:
@sudz123 Looks like the math import isn't used anywhere so you should just change the `np.`s.

sudz123 committed:
Autolev: Initialised numpy_import flag values

NikhilPappu commented:
You don't have to set the flag outside the initialiser. Look at the
include_pydy flag. It should work similarly.

On Mon 10 Sep, 2018, 8:36 PM Sudarshan Kamath, <notifications@github.com>
wrote:

> @NikhilPappu <https://github.com/NikhilPappu> since the if condition for
> the numpy import is evaluated in the initialiser, will just changing the
> value of the flag in a method which uses np. suffice? I am having
> difficulty understanding how the class is initialised(speaking wrt to my
> last commit).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sympy/sympy/pull/15210#issuecomment-419946917>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFGWJ1AkDB9XY_2B0tcMZxCTrCP8PGokks5uZoAAgaJpZM4WgXaQ>
> .
>

moorepants commented on sympy/parsing/autolev/__init__.py:
```diff
@@ -47,10 +47,9 @@ def parse_autolev(autolev_code, include_numeric=False, numpy_import=False):
     >>> my_al_text = '\\n'.join(my_al_text)
     >>> from sympy.parsing.autolev import parse_autolev
-    >>> print(parse_autolev(my_al_text, include_numeric=True))
+    >>> print(parse_autolev(my_al_text, include_numeric=True, numpy_import=True))
```
I thought the `include_numeric` flag added the numpy related code. I don't think we need an additional flag.

sudz123 replied:
So currently does the issue still exist? Isn't the numpy related code currently added regardless of any flag?

moorepants replied:
Some imports are not needed if include_numeric is false.

moorepants replied:
If `include_numeric` is False, there should be no numpy or pydy imports in the generated code. The only imports should be from sympy.

sudz123 committed:
autolev: set include_numeric as a flag for numpy imports

sudz123 commented:
I removed `numpy_import` and set `include_numeric` as a flag for numpy. @NikhilPappu so does this work? Should I proceed to re-write the test examples?

sudz123 commented:
Also while building the test examples, in `test_autolev.py` aren't we passing `include_numeric` as true always. In line : 
`generated_code = parse_autolev(f, include_numeric=True)`

moorepants commented:
You may need to disable the include_numeric for some tests. Only include if it needs numeric codes.

czgdp1807 commented:
@sudz123 Are you still working on this?

kenluck2001 commented:
Can I work on this PR?

czgdp1807 commented:
@kenluck2001 I think you should proceed with a new PR to complete this one, if the author doesn't respond for 72 hours.

kenluck2001 commented:
@czgdp1807 I have taken over. I will proceed to PR soon.

czgdp1807 commented:
Closing this PR in favour of https://github.com/sympy/sympy/pull/17753
Thank you all for the contributions. Please let me know if it is needed to be opened again for some reasons.
